lua/fspectate/cl_init.lua

```diff
@@ -29,6 +31,28 @@ local cam_Start3D = cam.Start3D
 local cam_End3D = cam.End3D
 local draw_WordBox = draw.WordBox
 
+-- set up convars for customization
+local convarTable = {
+    aimlines = { default = 1, func = function( val ) showBeams = val end },
+    xray = { default = 0, func = function( val ) showChams = val end },
+    crosshair = { default = 1, func = function( val ) showCrosshair = val end },
+    playerinfo = { default = 1, func = function( val ) showPlayerInfo = val end },
+    names = { default = 1, func = function( val ) showNames = val end },
+    health = { default = 0, func = function( val ) showHealth = val end },
+    weapon = { default = 0, func = function( val ) showWeaponName = val end },
+    rank = { default = 0, func = function( val ) showRank = val end },
+    e2s = { default = 0, func = function( val ) showE2s = val end },
+    sfs = { default = 0, func = function( val ) showSFs = val end },
+}
+
+for convarname, tbl in pairs( convarTable ) do
+    local convarString = "fspectate_" .. convarname
+    CreateClientConVar( convarString, tbl.default and 1 or 0, true, true )
+    cvars.AddChangeCallback( convarString, function( _, _, new )
+        tbl.func( tobool( new ) )
+    end )
+end
```

Comment:
```suggestion
local convarTable = {
    aimlines = true,
    xray = false,
    crosshair = true,
    playerinfo = true,
    names = true,
    health = false,
    weapon = false,
    rank = false,
    e2s = false,
    sfs = false,
}

for convarname, val in pairs( convarTable ) do
    local convarString = "fspectate_" .. convarname
    CreateClientConVar( convarString, val, true, true )
    cvars.AddChangeCallback( convarString, function( _, _, new )
        convarTable[convarName) = tobool( new )
    end )
end

setfenv( 0, setmetatable( convarTable, {
    __index = function( self, idx )
        local val = rawget( self, idx )
        if val != nil then return val end
        
        return _G[idx]
    end
} ) )
```

Lol I wonder if something like this would work - in theory you could remove the locals here. I highly doubt it'd be any more efficient than the current setup though, because it's doing an extra table lookup probably.

Reply:
Yeah, i feel like this would just overcomplicate it and have neglectable perfomance increase, also you used `!=` and `convarTable[convarName)` 🤨